website/integrations: add Actual budget (#12590)
goauthentik/authentik: website/integrations/services/actual-budget/index.mdx

---
title: Integrate with Actual Budget
sidebar_label: Actual Budget
---

# Actual Budget

<span class="badge badge--secondary">Support level: Community</span>

## What is Actual Budget

> Actual Budget is a super-fast and privacy-focused app for managing your finances. At its core is the well-proven and much-loved envelope budgeting methodology.
You own your data and can do whatever you want with it. It features multi-device sync, optional end-to-end encryption, and much more.
> -- https://actualbudget.org/
>
> This guide explains how to configure Actual Budget to use authentik as the OAuth provider for logging in to the Web GUI.

## Preparation

The following placeholders are used in this guide:

- `actual.company` is the FQDN of the Actual Budget install.
- `authentik.company` is the FQDN of the authentik install.

> ## authentik configuration

Author: Heya, good morning 😊 thank you for reviewing my PR again!

Is there an example or template on how the documentation pages should look like after we all should use the wizard instead of the manual steps?

Just oriented myself now the last PR for Semaphore UI 😂

> ## authentik configuration

Author: Is this mandatory for this PR? I would do it for my next guide, I don't want to start from scratch again 😭

Learned something new how git works and what I should try to avoid next time (node modules 😂 f.e.)

I guess this guide is close to the same structure as my other guides I already have written here an you reviewed, at least I hope so 😊

> ## authentik configuration

Reviewer: Let's leave it as it is for now, and change it to use the Wizard later. The Wizard is still in Preview, anyway... so when we make it the default then we can update all of the Integration guides. :-)

Thanks @nicedevil007 for this guide! Let me review quickly (thanks @4d62 for your edits!) and let's get this merged...

[Create](https://docs.goauthentik.io/docs/add-secure-apps/applications/manage_apps#add-new-applications) an OAuth2/OpenID provider and an application in authentik. Use the following parameters for the OAuth2/OpenID provider:

**Provider:**

- Name: `SP-actual`
- Client type: `Confidential`
- Redirect URIs/Origins (RegEx): `https://actual.company/openid/callback`

> - Signing Key: `authentik Self-signed Certificate`

Author: In discord of actual was a recent post about OpenID authentication and most guys got problems with right signing keys. The solution for them was to switch back to authentik's self signed cert.

Take note of the Client ID and Client Secret; you will need to provide them to Actual Budget in Step 3.

Leave the remaining values as default. Durations can be adjusted as needed.

**Application:**

- Name: `Actual Budget`
- Slug: `actual`
- Launch URL: `https://actual.company/`

## Actual Budget configuration

> 1. Log into your Actual Budget using a browser of your choice and open your budget.

Author: The guy got a list of all budgets and it just simple click on the name/entry of the budget. It is designed like a button with the name of the budget inside.

4. Enable the option `OpenID authentication method` by selecting the checkbox next to it.

5. Scroll up to the new option `Authentication method...` and click the `Start using OpenID` button.

6. Enter the variables provided by authentik here.  
    - Set `OpenID Provider` to `Authentik`  
    - Set `OpenID provider URL` to `https://authentik.company/application/o/actual/`  
    - Set `Client ID` to `<client-id>`  
    - Set `Client secret` to `<client-secret>`

:::info
The first user that logs into Actual Budget via OpenID will become the owner of the budget.
:::

## Test the login

- Open a browser of your choice and navigate to `https://actual.company`.
- Select the OpenID login method in the dropdown menu and click the `Sign in with OpenID` button.
- You should be redirected to authentik (with the login flows you created), and then authentik will redirect you back to the `https://actual.company` URL.
- If you are redirected back to the `https://actual.company` URL and can see the budget file selection page, the setup was successful.

:::info

> Users are not automatically created when logging in with authentik. The owner must manually create each user. To do so, click `Server online` at the top next to your name and select `User Directory`. Add a new user. The `Username` must match the one in authentik. You can now grant the new user access to your budget by clicking `Server online` next to your name at the top and selecting `User Access`.

Author: That is correct. First one is the king.

> Users are not automatically created when logging in with authentik. The owner must manually create each user. To do so, click `Server online` at the top next to your name and select `User Directory`. Add a new user. The `Username` must match the one in authentik. You can now grant the new user access to your budget by clicking `Server online` next to your name at the top and selecting `User Access`.

Author: That is correct. First one is the king.

---
title: Integrate with Actual Budget
sidebar_label: Actual Budget
---

# Actual Budget

<span class="badge badge--secondary">Support level: Community</span>

## What is Actual Budget

> Actual Budget is a super-fast and privacy-focused app for managing your finances. At its core is the well-proven and much-loved envelope budgeting methodology.
> You own your data and can do whatever you want with it. It features multi-device sync, optional end-to-end encryption, and much more.

> >

Reviewer: Normally I would rewrite the markety-hype description of the product, and state plainly what the product does, common use cases, etc. However, since this is a direct quote from the product's website, we can leave it for now. As a future project, we will probably rewrite all of them.

Author: I can do it for this PR as well, we don't have to wait 😂

> -- https://actualbudget.org/
>
> This guide explains how to configure Actual Budget to use authentik as the OAuth provider for logging in to the Web GUI.

## Preparation

The following placeholders are used in this guide:

- `actual.company` is the FQDN of the Actual Budget install.
- `authentik.company` is the FQDN of the authentik install.

## authentik configuration

[Create](https://docs.goauthentik.io/docs/add-secure-apps/applications/manage_apps#add-new-applications) an OAuth2/OpenID provider and an application in authentik. Use the following parameters for the OAuth2/OpenID provider:

**Provider:**

- Name: `SP-actual`
- Client type: `Confidential`
- Redirect URIs/Origins (RegEx): `https://actual.company/openid/callback`
- Signing Key: Select any available signing keys.

:::info
Actual Budget supports the RS256 algorithm. Please ensure to consider this when choosing the appropriate signing key.
:::

> Take note of the Client ID and Client Secret; you will need to provide them to Actual Budget in Step 3.

Reviewer: Yay, a semi-colon used properly! :-)

Author: ChatGPT helped me a bit 😎 as might already know, English is not my native language 😂

Leave the remaining values as default. Durations can be adjusted as needed.

**Application:**

- Name: `Actual Budget`
- Slug: `actual`
- Launch URL: `https://actual.company/`

## Actual Budget configuration

1. Sign in to Actual Budget with a browser of your choice and access your budget by clicking on its name.

> 2. Click your budget in the top-left corner to open the dropdown menu and select `Settings`.

Reviewer: ```suggestion
2. Click your budget in the top-left corner to open the dropdown menu and select **Settings**.
```

Reviewer: Yes, now I see what 4d62 was saying.... the UI components, the buttons and field names, etc., should be bolded, not in code format.

> 3. Scroll to the bottom and select `Show advanced settings`. Scroll again and select `I understand the risks, show experimental features`.

Reviewer: ```suggestion
3. Scroll to the bottom and select **Show advanced settings**. Scroll again and select **I understand the risks, show experimental features**.
```

> 4. To enable the option `OpenID authentication method` select the checkbox next to it.

Reviewer: ```suggestion
4. To enable the option **OpenID authentication method** select the checkbox next to it.
```

> 5. Scroll up to the new option `Authentication method...` and click the `Start using OpenID` button.

Reviewer: ```suggestion
5. Scroll up to the new option **Authentication method...** and click **Start using OpenID**.
```

> 6. Enter the variables provided by authentik here.

Author: the last part was added because you adviced me to add something that this part is not overread. So what should I do now. Maybe I missunderstood your comment from here? https://github.com/goauthentik/authentik/pull/12590#discussion_r1906181053

>     - Set `OpenID Provider` to `Authentik`

Reviewer: ```suggestion
    - Set **OpenID Provider** to `Authentik`
```

>     - Set **OpenID Provider** to `Authentik`

Author: Unfortunately it is writte with a capital A on their page and it is hardcoded :/ right now it is `Authentik`

![image](https://github.com/user-attachments/assets/c5bf3309-83fb-4a56-981c-d2164f0619ae)

>     - Set **OpenID Provider** to `Authentik`

Author: should I change it to `authentik` and raise an issue there :D

>     - Set `OpenID provider URL` to `https://authentik.company/application/o/actual/`

Reviewer: ```suggestion
    - Set **OpenID provider URL** to `https://authentik.company/application/o/actual/`
```

>     - Set `Client ID` to `<client-id>` to the value shown in the authentik provider.

Reviewer: ```suggestion
    - Set **Client ID** to `<client-id>` (the value shown in the authentik provider)
```

>     - Set `Client ID` to `<client-id>` from Step 2

Reviewer: ```suggestion
    - Set `Client ID` to `<client-id>`
```

Reviewer: Do you mean Step 2.. under the authentik configuration section (which is not numbered with steps) or ...?

Author: Yes but your suggestion is absolutely more understandable 💪

>     - Set `Client secret` to `<client-secret>` to the value shown in the authentik provider.

Reviewer: ```suggestion
    - Set **Client secret** to `<client-secret>` (the value shown in the authentik provider)
```

>     - Set `Client secret` to `<client-secret>` from Step 2

Reviewer: ```suggestion
    - Set `Client secret` to `<client-secret>` to the value shown in the authentik provider.
```

:::warning

> The first user to log into Actual Budget via OpenID will become the owner and administrator with the highest privileges for the budget. Check the info box at the bottom for more information on how to create additional users.

Reviewer: ```suggestion
The first user to log into Actual Budget via OpenID will become the owner and administrator with the highest privileges for the budget. For more information on how to create additional users, see the Note below.
```

:::

## Test the login

- Open a browser of your choice and navigate to `https://actual.company`.

> - Select the OpenID login method in the dropdown menu and click the **Sign in with OpenID** button.

Reviewer: ```suggestion
- Select the OpenID login method in the dropdown menu and click **Sign in with OpenID**.
```

- You should be redirected to authentik (with the login flows you created), and then authentik will redirect you back to the `https://actual.company` URL.
- If you are redirected back to the `https://actual.company` URL and can see the budget file selection page, the setup was successful.

:::info

> Users are not automatically created when logging in with authentik. The owner must manually create each user in Actual Budget. To do so, click `Server online` at the top next to your name and select `User Directory`. Add a new user. The `Username` must match the one in authentik. You can now grant the new user access to your budget by clicking `Server online` next to your name at the top and selecting `User Access`.

Reviewer: ```suggestion
Users are not automatically created when logging in with authentik. The owner must manually create each user in Actual Budget. To do so, click **Server online** at the top next to your name and select **User Directory**. Add a new user. The `Username` must match the one in authentik. You can now grant the new user access to your budget by clicking **Server online** next to your name at the top and selecting **User Access**.
```

> Users are not automatically created when logging in with authentik. The owner must manually create each user. To do so, click `Server online` at the top next to your name and select `User Directory`. Add a new user. The `Username` must match the one in authentik. You can now grant the new user access to your budget by clicking `Server online` next to your name at the top and selecting `User Access`.

Reviewer: ```suggestion
Users are not automatically created when logging in with authentik. The owner must manually create each user in Actual Budget. To do so, click `Server online` at the top next to your name and select `User Directory`. Add a new user. The `Username` must match the one in authentik. You can now grant the new user access to your budget by clicking `Server online` next to your name at the top and selecting `User Access`.
```
